Library/Formula/libgraphqlparser.rb

```diff
@@ -0,0 +1,206 @@
+                      "name": {
+                        "kind": "Name",
+                        "loc": {
+                          "start": 5,
+                          "end": 9
+                        },
+                        "value": "name"
+                      },
+                      "arguments": null,
+                      "directives": null,
+                      "selectionSet": null
+                    },
+                    {
+                      "kind": "Field",
+                      "loc": {
+                        "start": 5,
+                        "end": 8
+                      },
+                      "alias": null,
+                      "name": {
+                        "kind": "Name",
+                        "loc": {
+                          "start": 5,
+                          "end": 8
+                        },
+                        "value": "age"
+                      },
+                      "arguments": null,
+                      "directives": null,
+                      "selectionSet": null
+                    },
+                    {
+                      "kind": "Field",
+                      "loc": {
+                        "start": 5,
+                        "end": 6
+                      },
+                      "alias": null,
+                      "name": {
+                        "kind": "Name",
+                        "loc": {
+                          "start": 5,
+                          "end": 12
+                        },
+                        "value": "friends"
+                      },
+                      "arguments": null,
+                      "directives": null,
+                      "selectionSet": {
+                        "kind": "SelectionSet",
+                        "loc": {
+                          "start": 13,
+                          "end": 6
+                        },
+                        "selections": [
+                          {
+                            "kind": "Field",
+                            "loc": {
+                              "start": 7,
+                              "end": 11
+                            },
+                            "alias": null,
+                            "name": {
+                              "kind": "Name",
+                              "loc": {
+                                "start": 7,
+                                "end": 11
+                              },
+                              "value": "name"
+                            },
+                            "arguments": null,
+                            "directives": null,
+                            "selectionSet": null
+                          }
+                        ]
+                      }
+                    }
+                  ]
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+    EOS
+
+    json_ast = JSON.parse(sample_ast)
```

Comment:
Unless we need `sample_ast` later in the test we should probably set the variable directly:

```rb
json_ast = {
  "kind" => "Document",
  "loc" => {
    "start" => 1,
    "end" => 2,
    ...
  }
  ...
}
```